explainer.md

```diff
@@ -36,193 +97,126 @@ self.addEventListener('launch', event => {
-    document, but if the user clicks a link to a document that is already open
-    in a tab, focus that tab instead of opening a duplicate.
-  * Keep in mind, in all of these cases, that the user can override this with
-    "Open in new tab" if they really want two tabs for the same app / document.
-* **Advanced protocol handlers**. Sites that use
-  [`registerProtocolHandler`](https://html.spec.whatwg.org/multipage/system-state.html#dom-navigator-registerprotocolhandler)
-  to handle URLs of a certain scheme may wish to handle incoming URLs without
-  opening a new tab.
-  * e.g., (An extension of the previous point): A mail client already has an
-    open tab; if a "`mailto:`" URL is clicked, it focuses the existing open tab,
-    opening a Compose sub-window, rather than opening a new tab to compose the
-    email.
-  * e.g., A "`magnet:`" URL is handled by a torrent client, which automatically
-    starts downloading the file, showing a notification but not opening a new
-    window or tab.
-
-## Use cases (future)
-
-* **Single-window installable apps**. The same use case as "single-tab web apps"
-  above, but for installed apps that run in a window. With a focus on
-  [installable web apps](https://www.w3.org/TR/appmanifest/) that "look and
-  feel" more like native apps, this use case makes a lot more sense, as
-  single-window apps are very common on both mobile and desktop platforms.
-* **Web Share Target API**. Similar to `registerProtocolHandler`, share targets
-  may wish to deal with an incoming piece of shared data without opening a new
-  tab. One particular use case is a "save for later" app; when it receives a
-  share, it just shows a notification "Saved", rather than opening a new tab or
-  window. Note that our original Share Target design ([Approach 2 in this
-  explainer](https://github.com/WICG/web-share-target/blob/master/docs/explainer.md#sample-code))
-  involved a new service worker event, which would be obsoleted by the `launch`
-  event.
-* **Deep-link shortcuts API**. A proposed API ([proposal
-  1](https://gist.github.com/kenchris/0acec2790cd38dfdff0a7197ff00d1de);
-  [proposal
-  2](https://docs.google.com/a/chromium.org/document/d/1WzpCnpc1N7WjDJnFmj90-Z5SALI3cSPtNrYuH1EVufg/edit))
-  to allow web apps to create shortcuts to deep links within the app. The
-  original proposal was to have those links fire a new type of service worker
-  event. The second proposal just navigates to a URL, a simpler but less
-  powerful approach. With the `launch` event, the latter approach would be just
-  as powerful as the former.
-
-# Background
+### Event Definition
 
-* [Service Worker GitHub issue](https://github.com/w3c/ServiceWorker/issues/1028)
-* [mgiuca proposal document](https://docs.google.com/document/d/1jWLpNEFttyLTnxsHs15oT-Hn8I81N0cwUa3JjISoPV8/edit)
+```ts
+interface LaunchEvent : ExtendableEvent {
+  readonly attribute Request request;
+  readonly attribute DOMString? clientId;
```

Comment:
I wonder why mgiuca wanted to include the clientId? The main reason I can see for including it is a situation like this:
-User clicks a navigation in an existing tab which triggers a LE
-The default browser behaviour (of this particular browser) is to open the link in a new window
-The App wants the browser to open the link in the existing tab instead of a new window

In this case, the clientId could presumably be used to see what tab the navigation came from and handle it there. I don't know the practical use case for this, but just a thought.

In any case, I don't think we need client ID for the MVP. We could easily add it later if use cases justified it. I've removed it from the explainer.

Reply:
Cool :) Thinking about it, the site can probably get some idea of where the navigation came from by looking at the referrer header. Maybe that, coupled with a completely opaque clientId would be enough for them to decide.

That said, I don't think we want to get into a situation where sites can do different things based on who navigated to them, it'd break my mental model of the web